doc/visualizer-in-depth.md

```diff
@@ -0,0 +1,524 @@
+classes: {
+  init: {style: {
+     stroke: red
+     font-size: 18
+     font-color: darkred
+  } }
+  parChg: {style: {
+     stroke: orange
+     font-size: 18
+     font-color: chocolate
+  } }
+  seqChg: {style: {
+     stroke: green
+     font-size: 18
+     font-color: darkGreen
+  } }
+  resize: {style: {
+     stroke: purple
+     font-size: 18
+     font-color: indigo
+  } }
+  any: {style: {
+     stroke: black
+     stroke-width: 2
+     font-size: 18
+     font-color: black
+  } }
+  dot: {
+    shape: circle
+    width: 10
+  }
+  method: {
+    height: 40
+    style: {
+      border-radius: 10
+      font-size: 20
+    }
+  }
+  returns: {
+    shape: cf-many
+    style.opacity: 0.0
+  }
+  deflt: {
+    style: {
+      stroke-width: 3
+      stroke-dash: 5
+    }
+  }
+  pseudolabel: {
+    shape: text
+    style: {
+      italic: true
+      font-size: 16
+    }
+  }
+}
+
+secret: {
+   label: ""
+   near: center-left
+   style: {opacity: 0.0}
+   Start: "" {
+      class: [dot; init]
+      style.fill: red
+   }
+   dummy: "" {
+      height: 1115
+      style.opacity: 0.0
+   }
+   Start -> dummy {style.opacity: 0.0}
+}
+
+early: {
+  direction: right
+  label: "can't use\nsequence or sketch       "
+  label.near: top-right
+  style.fill: "#fee"
+  style.font-size: 18
+  Init: |md
+    ### Create visualizer object
+    ```txt
+    property initializers,
+    constructor
+    ```
+  |
+  SPC: |md
+    ### Parameter checks
+    ```txt
+         individual
+    validate() functions
+    ```
+  |
+  Init -> SPC: {class: init}
+}
+
+secret.Start -> early.Init: "init (page load/\nvisualizer change)" {
+   class: init
+   style.font-size: 20
+}
+
+mid: {
+  label: "    sequence available,\nno sketch"
+  label.near: top-left
+  style.fill: "#ffe"
+  style.font-size: 18
+  CheckPar: "checkParameters()" {class: method}
+  parUp: "parameter\nchange" {class: [pseudolabel; parChg]}
+  checkPar -- parUp {class: parChg}
+  ParChg: "parametersChanged()" {class: method}
+  parUp -> ParChg {class: parChg}
+  Reset: "reset()" {class: method}
+  checkPar --> Reset: "init" {class: init}
+  checkPar --> Reset: "sequence\nchange" {class: seqChg}
+  ParChg -> Reset {class: [deflt; parChg]}
+  PreSk: "presketch()" {class: method}
+  Reset -> PreSk: "init" {class: init}
+  Reset -> PreSk: "sequence\nchange" {class: seqChg}
+  Reset -> PreSk: "canvas\nresize" {class: resize}
+  routingQ: "" {class: [dot; resize]; width: 2}
+  routingR: "" {class: [dot; resize]; width: 2}
+  routingQ -- routingR -> Reset: {class: [deflt; resize]}
+  routingZ: "" {class: [dot; any]; width: 1}
+  PreSk <-- routingZ {
+     class: any
+     source-arrowhead: "in any\nevent" {shape: cf-many}
+  }
+}
+early.SPC -> mid.CheckPar: "init" {class: init}
+early.SPC -> mid.CheckPar: "parameter\nchange" {class: parChg}
+late: {
+  near: center-right
+  direction: up
+  label: "sequence & sketch available"
+  label.near: top-right
+  style.fill: "#efe"
+  style.font-size: 18
```

Comment:
Weirdly, I swear the font on this box "sequence & sketch available" is smaller than for the other boxes... but I can't see why from the code ... not a big deal
![Screenshot from 2024-11-13 17-14-49](https://github.com/user-attachments/assets/be96935f-9266-4b4f-b0c9-cacd9ecf3750)

Reply:
I agree that it looks better bumped up one point -- maybe because it's in a much bigger empty "field". Will do.